core/src/main/java/de/danoeh/antennapod/core/service/playback/PlaybackService.java

```diff
@@ -1815,15 +1817,59 @@ public void onSetPlaybackSpeed(float speed) {
             setSpeed(speed);
         }
 
+        private boolean playStateBeforeClick = false;
+        private long lastClickTime = 0;
+        private Handler clickHandler = new Handler(Looper.getMainLooper());
+        private int clickCount = 0;
+
+        private Runnable clickRunnable = new Runnable() {
+            @Override
+            public void run() {
+                if (clickCount == 1) {
+                    if (playStateBeforeClick) {
+                        onPause();
+                    } else {
+                        onPlay();
+                    }
+                } else if (clickCount == 2) {
+                    onFastForward();
+                } else if (clickCount == 3) {
+                    onRewind();
+                    onPlay(); // otherwise playback is paused?
+                }
+                // Reset
+                clickCount = 0;
+            }
+        };
+
         @Override
         public boolean onMediaButtonEvent(final Intent mediaButton) {
             Log.d(TAG, "onMediaButtonEvent(" + mediaButton + ")");
             if (mediaButton != null) {
-                KeyEvent keyEvent = mediaButton.getParcelableExtra(Intent.EXTRA_KEY_EVENT);
-                if (keyEvent != null &&
-                        keyEvent.getAction() == KeyEvent.ACTION_DOWN &&
-                        keyEvent.getRepeatCount() == 0) {
-                    return handleKeycode(keyEvent.getKeyCode(), false);
+                final KeyEvent keyEvent = mediaButton.getParcelableExtra(Intent.EXTRA_KEY_EVENT);
+                if (keyEvent != null && keyEvent.getRepeatCount() == 0) {
+                    // Check if the key event is a play/pause button to avoid other button presses
+                    if (keyEvent.getKeyCode() == KeyEvent.KEYCODE_MEDIA_PLAY_PAUSE
+                            || keyEvent.getKeyCode() == KeyEvent.KEYCODE_HEADSETHOOK) {
+                        if (keyEvent.getAction() == KeyEvent.ACTION_DOWN) {
+                            long clickTime = System.currentTimeMillis();
+                            long elapsedTime = clickTime - lastClickTime;
+                            if (elapsedTime <= 500) { // Assuming 500 milliseconds threshold for multiple clicks
+                                clickCount++;
+                            } else {
+                                clickCount = 1; // if more than 500ms has passed, reset to 1
+                            }
+                            clickHandler.removeCallbacks(clickRunnable); // always remove callbacks
+                            // always post a delayed clickRunnable.
+                            // It will get executed if no further clicks are detected within the threshold.
+                            clickHandler.postDelayed(clickRunnable, 500);
```

Comment:
This means that all play button actions are now 500ms slower. I haven't tried this, but I feel like this would be a bit annoying for users who don't use this feature (especially users who have headphones with double-click functionality that have some delay on top). I think 250ms should be enough

Reply:
After testing in 50ms increments from 250-600, 400ms seemed like the best balance between delay and capturing multi click events. For reference, pocketcasts uses a 600ms timer for double click events.